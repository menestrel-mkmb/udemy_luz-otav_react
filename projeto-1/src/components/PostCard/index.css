.post {
    background-color: #fff;
    box-shadow: 0 0 10px rgba( 0, 0, 0, .2 );
  
    width: clamp(280px, 100%, 350px);
  
    transition: transform .3s ease-in-out;
  }
  
  .post:hover {
    scale: 1.05;
  }
  
  .post__img {
    height: aspect-ratio( 16 / 9 );
    width: clamp(280px, 100%, 350px);
  
    margin: 0 auto;
  }
  
  .post__content {
    font-size: clamp(.5rem,1rem,2rem);
  
    padding: .5rem 1rem;
  }
  
  .post__title {
    font-size: clamp(1rem,1.5rem,3rem);
  }