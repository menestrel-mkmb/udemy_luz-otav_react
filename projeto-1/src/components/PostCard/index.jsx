import './styles.css';

export default function PostCard(props) {
    const { post } = props;

    return (
    <div className='post card'>
        <img
            className='post__img'
            src={post.url}
            alt={post.title}
        />
        <div
            className='post__content'
        >
            <h1
                className='post__title'
            >
                {post.title}
            </h1>
            <p
                className='post__body'
            >
                {post.body}
            </p>
        </div>
    </div>)
}