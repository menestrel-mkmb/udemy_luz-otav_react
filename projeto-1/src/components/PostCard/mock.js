export const postCardMockProps = {
    id: 1,
    title: 'TÃ­tulo 1',
    body: 'Lorem1',
    url: 'img/img.jpg'
};