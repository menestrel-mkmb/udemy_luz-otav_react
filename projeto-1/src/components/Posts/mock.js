export const postsMockProps = [
    {
        id: 1,
        title: 'Título 1',
        body: 'Lorem1',
        url: 'img/img1.jpg'
    },
    {
        id: 2,
        title: 'Título 2',
        body: 'Lorem2',
        url: 'img/img2.jpg'
    },
    {
        id: 3,
        title: 'Título 3',
        body: 'Lorem3',
        url: 'img/img3.jpg'
    }
];