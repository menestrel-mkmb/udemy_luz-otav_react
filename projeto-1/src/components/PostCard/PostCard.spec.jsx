import { render, screen } from "@testing-library/react";
import PostCard from ".";
import { postCardMockProps } from "./mock";

const props = postCardMockProps;

describe("<PostCard />", () => {
    it('should render PostCard', () => {

        render(<PostCard post={props} />);

        expect(screen.getByRole('img', { name: 'Título 1' }))
            .toHaveAttribute('src', 'img/img.jpg');
        expect(screen.getByRole('img', { name: 'Título 1' }))
            .toHaveAttribute('alt', 'Título 1');
        expect(screen.getByRole('heading', { name: 'Título 1' }))
            .toBeInTheDocument();
        expect(screen.getByText('Lorem1'))
            .toBeInTheDocument();
    });

    it('should match snapshot', () => {
        const { container } = render(<PostCard post={props} />);
        expect(container.firstChild).toMatchSnapshot();
    });
});