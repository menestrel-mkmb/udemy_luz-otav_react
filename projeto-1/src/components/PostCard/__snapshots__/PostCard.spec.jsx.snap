// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> should match snapshot 1`] = `
<div
  class="post card"
>
  <img
    alt="Título 1"
    class="post__img"
    src="img/img.jpg"
  />
  <div
    class="post__content"
  >
    <h1
      class="post__title"
    >
      Título 1
    </h1>
    <p
      class="post__body"
    >
      Lorem1
    </p>
  </div>
</div>
`;
